server/py/services/api/initial_data.py

```diff
@@ -877,40 +877,84 @@ def _perform_version_8_data_migrations(
 def _perform_version_9_data_migrations(
     db: framework.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
 ):
-    _migrate_function_kind(db, db_session, chunk_size=500)
+    _migrate_data(
+        db,
+        db_session,
+        framework.db.sqldb.models.Function,
+        "kind",
+        _handle_function_kind,
+    )
+    _migrate_data(
+        db,
+        db_session,
+        framework.db.sqldb.models.ArtifactV2,
+        "producer_uri",
+        _handle_artifact_producer_uri,
+    )
```

Comment:
I think it is now too general and you can't understand what is the migration doing.
There are 2 options:
1. have another wrapper function for each (e.g. `_ensure_function_kind` and `_add_producer_uri_to_artifact` that will call `_migrate_data`
2. Add comments on what each call does.

Although the first option seems like an overkill I think I prefer it for readability and functionality, but your choice.